x-pack/plugins/security_solution/server/lib/detection_engine/fleet_integrations/api/get_installed_integrations/installed_integration_set.ts

```diff
@@ -49,7 +49,7 @@ export const createInstalledIntegrationSet = (): IInstalledIntegrationSet => {
     }
 
     // Actual `installed_version` is buried in SO, root `version` is latest package version available
-    const installedPackageVersion = fleetPackage.savedObject.attributes.install_version;
+    const installedPackageVersion = fleetPackage?.savedObject?.attributes?.install_version || '';
```

Comment:
Why `savedObject` and its `attributes` became optional? I don't see any changes to the `PackageClient` or I just miss something.

Btw `fleetPackage`'s type here says it's not optionals. If it can be `undefined` in runtime the type has to be fixed.

Reply:
You're totally right about `attributes`, it should never be optional and in fact I fixed it everywhere.

However,  `savedObject` can be optional in some cases. I'll explain what I found.

`fleetPackage` is a `PackageListItem` type and comes from Fleet [getPackages](https://github.com/elastic/kibana/blob/9156d89eb50cbfce58f253c3cd919948f903deb3/x-pack/plugins/fleet/server/services/epm/packages/get.ts#L116-L124) function. The `savedObject` is conditionally added by [createInstallableFrom](https://github.com/elastic/kibana/blob/9156d89eb50cbfce58f253c3cd919948f903deb3/x-pack/plugins/fleet/server/services/epm/packages/index.ts#L45) so the SO could be undefined in some cases. 
I'm not sure why with the previous definition it was never optional, but as soon as I replaced the type the linter failed in several places so I dug in the code and found this function. 

I checked the usage of `createInstalledIntegrationSet`. `addPackage` is used in https://github.com/elastic/kibana/blob/9156d89eb50cbfce58f253c3cd919948f903deb3/x-pack/plugins/security_solution/server/lib/detection_engine/fleet_integrations/api/get_installed_integrations/route.ts#L38-L45  
where `allThePackages` come from `fleet.packages.getPackages()`, so that's the same type as above.